networkx/algorithms/tree/coding.py

```diff
@@ -312,7 +312,7 @@ def parents(u):
         if v < index and degree[v] == 1:
             u = v
         else:
-            index = u = min(k for k in range(index + 1, n) if degree[k] == 1)
```

Comment:
The result of this line is always the smallest k satisfying degree[k] == 1, but min doesn't know this and considers others, even after the first satisfying k has been found.